### PR TITLE
[#4] Create Docker for Development

### PR DESCRIPTION
**Pull request by mappuji**
@dogi and @paulbert we need to arrange a demo session for this one.

Basically in this PR I want to increase the development environment for faster development using docker and docker-compose. Checkout my PR and see readme for testing.

Solved #4

### CODE REVIEW

File: docker-dev/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Docker for your development environment
+
+## Description
+This docker compose can be use for your development environment and very handy, you can spawn the development environment in a matter of seconds and start your development. Your code changes in host folder are automatically reflected to docker and ready to test in your browser.
+
+## How to use
```
Comment (mappuji) [inline docker-dev/README.md]: Add how to use it

tells about docker logs

File: docker-dev/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+services:
+  couchdb:
+    expose:
+      - 5984
+    image: klaemo/couchdb
```
Comment (mappuji) [inline docker-dev/docker-compose.yml]: Better to use official docker image I think

File: docker/planet/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Add CORS to CouchDB so app has access to databases
+git clone https://github.com/pouchdb/add-cors-to-couchdb.git
+cd add-cors-to-couchdb
+npm install
+node bin.js http://couchdb:5984
+cd -
+# End add CORS to CouchDB
+
+# Add initial Couch databases here
```
Comment (dogi) [inline docker/planet/docker-entrypoint.sh]: remove trailing and following lines ;) thanks @mappuji

File: docker/db-init/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+#CORS
+git clone https://github.com/pouchdb/add-cors-to-couchdb.git
+cd add-cors-to-couchdb
+npm install
+node bin.js http://couchdb:5984
```
Comment (dogi) [inline docker/db-init/docker-entrypoint.sh]: move to after wait for couchdb

File: docker/db-init/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+cd -
+
+#WAIT_TIME
+echo  "Waiting for couchdb to start"
+WAIT_TIME=0
+until curl couchdb:5984 || [ $WAIT_TIME -eq 180 ]; do
+echo "..."
+sleep 5
+WAIT_TIME=$(expr $WAIT_TIME + 5)
+done
```
Comment (dogi) [inline docker/db-init/docker-entrypoint.sh]: check if couchdb is already initialized

File: docker/planet/docker-entrypoint.sh (hunk 3 above)
Comment (dogi) [reply docker/planet/docker-entrypoint.sh]: or just remove it ...

File: docker/planet/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  # install angular cli
+  && npm install -g @angular/cli
+
+# exposes port 3000 but your app may use any port specified in it
+EXPOSE 3000
+
+VOLUME ["/$APP"]
+WORKDIR /$APP
+
+ADD ./docker-entrypoint.sh docker-entrypoint.sh
```
Comment (dogi) [inline docker/planet/Dockerfile]: ;)

**Pull request merged**